src/parseFromAnchor.js

```diff
@@ -12,7 +26,7 @@ function parseFromAnchor(opt_uri) {
 		hostname: link.hostname,
 		password: link.password,
 		pathname: link.pathname[0] === '/' ? link.pathname : '/' + link.pathname,
-		port: link.port,
+		port: extractPortFromURI(opt_uri),
```

Comment:
Hey, guys.

This method might be weird but it's to prevent a Safari 5 behaviour that has an unexpected default port value '0'.
"http://localhost" give us "0" as the port number;